pkg/cmd/experimental/syncgroups/cli/prunegroups.go

```diff
@@ -0,0 +1,233 @@
+package cli
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	kerrs "k8s.io/kubernetes/pkg/util/errors"
+
+	"github.com/openshift/origin/pkg/auth/ldaputil"
+	"github.com/openshift/origin/pkg/auth/ldaputil/ldapclient"
+	osclient "github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups"
+	"github.com/openshift/origin/pkg/cmd/server/api"
+	"github.com/openshift/origin/pkg/cmd/server/api/validation"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const (
+	PruneGroupsRecommendedName = "prune-groups"
+
+	pruneGroupsLong = `
+Prune OpenShift Groups referencing missing records on from an external provider.
+
+In order to prune OpenShift Group records using those from an external provider, determine which Groups you wish
+to prune. For instance, all or some groups may be selected from the current Groups stored in OpenShift that have
+been synced previously. Any combination of a literal whitelist, a whitelist file and a blacklist file is supported.
+The path to a sync configuration file that was used for syncing the groups in question is required in order to 
+describe how data is requested from the external record store. Default behavior is to prune all OpenShift groups 
+for which the external record does not exist.
+`
+	pruneGroupsExamples = `  # Prune all orphaned groups 
```

Comment:
don't all these simply display.  I would expect `--confirm` is needed to mutate.